reactor-netty-core/src/main/java/reactor/netty/resources/ConnectionProvider.java

```diff
@@ -488,6 +492,18 @@ public final Builder forRemoteHost(SocketAddress remoteHost, Consumer<HostSpecif
 			return this;
 		}
 
+		/**
+		 * Specifies the maximum number of connection pools that the provider can create.
+		 * If the number of connection pools created exceeds this value, a warning message is logged.
+		 * The value must be positive or zero; otherwise, the connection pools check is ignored.
```

Comment:
Why do we want to allow this value to be set to `0`? If one doesn't want pooling, one can disable the pooling.

Reply:
no particular reason really. it's a leftover from the original "expected connection pool" setting. as in, i expect 0 connection pool and show me a warning message if one is created. 

I will make all the suggested changes.